### PR TITLE
[Code Health]: Move proto and types related code to `proto/types`

### PR DESCRIPTION
**Pull request by red-0ne**
## Summary

Move the `proto` generated files and their dependencies from `x/<module>/types` to `proto/types/<module>`.

This PR is relatively big but does not introduce any new logic. It is mainly about moving files and renaming imports.

When approved, a corresponding `shannon-sdk` PR will be submitted.

## Issue

- #664 

## Type of change

Select one or more:

- [ ] New feature, functionality or library
- [ ] Bug fix
- [x] Code health or cleanup
- [ ] Documentation
- [ ] Other (specify)

## Testing

**Documentation changes** (only if making doc changes)
- [ ] `make docusaurus_start`; only needed if you make doc changes

**Local Testing** (only if making code changes)
- [x] **Unit Tests**: `make go_develop_and_test`
- [x] **LocalNet E2E Tests**: `make test_e2e`
- See [quickstart guide](https://dev.poktroll.com/developer_guide/quickstart) for instructions

**PR Testing** (only if making code changes)
- [ ] **DevNet E2E Tests**: Add the `devnet-test-e2e` label to the PR.
    - **THIS IS VERY EXPENSIVE**, so only do it after all the reviews are complete.
    - Optionally run `make trigger_ci` if you want to re-trigger tests without any code changes
    - If tests fail, try re-running failed tests only using the GitHub UI as shown [here](https://github.com/pokt-network/poktroll/assets/1892194/607984e9-0615-4569-9452-4c730190c1d2)

## Sanity Checklist

- [x] I have tested my changes using the available tooling
- [x] I have commented my code
- [x] I have performed a self-review of my own code; both comments & source code
- [ ] I create and reference any new tickets, if applicable
- [ ] I have left TODOs throughout the codebase, if applicable

### CODE REVIEW

File: pkg/appgateserver/endpoint_selector.go [hunk 1]
```diff
@@ -6,7 +6,7 @@ import (
 
 	shannonsdk "github.com/pokt-network/shannon-sdk"
 
-	sharedtypes "github.com/pokt-network/poktroll/x/shared/types"
+	sharedtypes "github.com/pokt-network/poktroll/proto/types/shared"
```
Comment (bryanchriswhite) [inline pkg/appgateserver/endpoint_selector.go:9]: Unalias

File: pkg/appgateserver/error_reply.go [hunk 2]
```diff
@@ -5,7 +5,7 @@ import (
 
 	sdktypes "github.com/pokt-network/shannon-sdk/types"
 
-	sharedtypes "github.com/pokt-network/poktroll/x/shared/types"
+	sharedtypes "github.com/pokt-network/poktroll/proto/types/shared"
```
Comment (bryanchriswhite) [inline pkg/appgateserver/error_reply.go:8]: Unalias

File: e2e/tests/init_test.go [hunk 3]
```diff
@@ -31,13 +31,13 @@ import (
 	"github.com/pokt-network/poktroll/pkg/client/block"
 	"github.com/pokt-network/poktroll/pkg/client/events"
 	"github.com/pokt-network/poktroll/pkg/client/tx"
+	apptypes "github.com/pokt-network/poktroll/proto/types/application"
+	prooftypes "github.com/pokt-network/poktroll/proto/types/proof"
+	sessiontypes "github.com/pokt-network/poktroll/proto/types/session"
+	sharedtypes "github.com/pokt-network/poktroll/proto/types/shared"
+	suppliertypes "github.com/pokt-network/poktroll/proto/types/supplier"
 	"github.com/pokt-network/poktroll/testutil/testclient"
```
[the remark this answers is not in the document]
Comment (bryanchriswhite) [reply e2e/tests/init_test.go:39]: You can probably use a regex to substitute: `\w+types\s+(github.com/pokt-network/poktroll/proto/types/)` for its capture group.

**Review by bryanchriswhite — APPROVED** [review]
Preemptively approving. As discussed, let's make sure we add a page/section to the "contributing" docs regarding checking the `option go_package` of newly generated .proto files when using ignite CLI to match the new convention.

**Comment by Olshansk** [comment]
@red-0ne So do we have to continue with the existing approach and throw this away completely?

**Comment by red-0ne** [comment]
> @red-0ne So do we have to continue with the existing approach and throw this away completely?

Yes, that's my take. I believe this should be closed for now and see if @bryanchriswhite has strong opinion on it.

It's just a refactor and we can still revive it if we think it's worth it.

**Comment by Olshansk** [comment]
Converting to draft until we decide on next steps here.

**Comment by Olshansk** [comment]
Closing this. Please reopen if needed

**Pull request closed without merging**